packages/core/src/getNextSnapshot.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  AnyActorLogic,
+  AnyActorScope,
+  EventFromLogic,
+  SnapshotFrom,
+  createEmptyActor
+} from '.';
+
+/**
+ * Determines the next snapshot for the given `actorLogic` based on
+ * the given `snapshot` and `event`.
+ *
+ * If the `snapshot` is `undefined`, the initial snapshot of the
+ * `actorLogic` is used.
+ * 
+ * @example
+  ```ts
+  import { getNextSnapshot } from 'xstate';
+  import { trafficLightMachine } from './trafficLightMachine.ts';
+
+  const nextSnapshot = getNextSnapshot(
+    trafficLightMachine, // actor logic
+    undefined, // snapshot (or initial state if undefined)
+    { type: 'TIMER' }); // event object
+  
+  console.log(nextSnapshot.value);
+  // => 'yellow'
+
+  const nextSnapshot2 = getNextSnapshot(
+    trafficLightMachine, // actor logic
+    nextSnapshot, // snapshot
+    { type: 'TIMER' }); // event object
+
+  console.log(nextSnapshot2.value);
+  // =>'red'
+  ```
+ */
+export function getNextSnapshot<T extends AnyActorLogic>(
+  actorLogic: T,
+  snapshot: SnapshotFrom<T> | undefined,
```

Comment:
Would it make sense to position the `snapshot` parameter last, so that you can omit it at the call site instead of passing `undefined`? That is: `getNextSnapshot(machine, { type: 'TIMER' })` instead of `getNextSnapshot(machine, undefined, { type: 'TIMER' })`.

Reply:
I don't think this would be beneficial. Almost all our methods accept `snapshot` before the `event`. Making an exception here would be odd. It also "forces" you to make some decision about this `snapshot` - optional parameters are so easy to forget about that bugs often are caused by people just forgetting to pass them in.

Reply:
I wasn’t aware of your ordering convention. And if it’s preferable to have people make a conscious decision about `snapshot`, then it all makes sense as is 👍